packages/protocol/test/utils/commit.ts

```diff
@@ -23,10 +27,49 @@ const commitBlock = async (
 ): Promise<{
     tx: ethers.ContractTransaction;
     commit: { hash: string; txListHash: string };
+    blockCommittedEvent: BlockCommittedEvent | undefined;
+    receipt: ethers.ContractReceipt;
 }> => {
     const commit = generateCommitHash(block);
     const tx = await taikoL1.commitBlock(commitSlot, commit.hash);
-    return { tx, commit };
+    const receipt = await tx.wait(1);
+    const blockCommittedEvent = receipt.events!.find(
+        (e) => e.event === "BlockCommitted"
+    ) as any as BlockCommittedEvent;
+    return { tx, commit, blockCommittedEvent, receipt };
 };
 
-export { generateCommitHash, commitBlock };
+const commitAndProposeLatestBlock = async (
+    taikoL1: TaikoL1,
+    l1Signer: any,
+    l2Provider: ethers.providers.JsonRpcProvider,
+    commitSlot: number = 0
+) => {
+    const { commitConfirmations } = await taikoL1.getConfig();
+    const block = await l2Provider.getBlock("latest");
+    const { tx, commit } = await commitBlock(
+        taikoL1.connect(l1Signer),
+        block,
+        commitSlot
+    );
+    const commitReceipt = await tx.wait(1);
+
+    for (let i = 0; i < commitConfirmations.toNumber(); i++) {
+        await sendTinyEtherToZeroAddress(l1Signer);
+    }
+
+    const proposeReceipt = await proposeBlock(
+        taikoL1.connect(l1Signer),
+        block,
+        commit.txListHash,
+        commitReceipt.blockNumber as number,
+        block.gasLimit,
+        commitSlot
+    );
+    const proposedEvent: BlockProposedEvent = (
```

Comment:
Do we always assume when we call this function, the `proposeBlock` tx always succeed? (I think not, for example, in test `should commit and be able to propose for all available slots, then revert when all slots are taken`), therefore, getting events from `proposeReceipt` will fail!

Reply:
same comment for propose.ts